sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/kafka/AbstractAzureKafkaOAuth2AutoConfigurationTests.java

```diff
@@ -0,0 +1,82 @@
+import static com.azure.spring.cloud.autoconfigure.implementation.kafka.AzureKafkaAutoconfigurationUtils.SASL_LOGIN_CALLBACK_HANDLER_CLASS_OAUTH;
+import static com.azure.spring.cloud.autoconfigure.implementation.kafka.AzureKafkaAutoconfigurationUtils.SASL_MECHANISM_OAUTH;
+import static com.azure.spring.cloud.autoconfigure.implementation.kafka.AzureKafkaAutoconfigurationUtils.SECURITY_PROTOCOL_CONFIG_SASL;
+import static org.apache.kafka.clients.CommonClientConfigs.SECURITY_PROTOCOL_CONFIG;
+import static org.apache.kafka.common.config.SaslConfigs.SASL_JAAS_CONFIG;
+import static org.apache.kafka.common.config.SaslConfigs.SASL_LOGIN_CALLBACK_HANDLER_CLASS;
+import static org.apache.kafka.common.config.SaslConfigs.SASL_MECHANISM;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public abstract class AbstractAzureKafkaOAuth2AutoConfigurationTests {
+    protected static final String SPRING_BOOT_KAFKA_PROPERTIES_PREFIX = "spring.kafka.properties.";
+    protected static final String SPRING_BOOT_KAFKA_PRODUCER_PROPERTIES_PREFIX = "spring.kafka.producer.properties.";
+    protected static final String SPRING_CLOUD_STREAM_KAFKA_PROPERTIES_PREFIX = "spring.cloud.stream.kafka.binder.configuration.";
+    protected static final String SPRING_CLOUD_STREAM_KAFKA_CONSUMER_PROPERTIES_PREFIX = "spring.cloud.stream.kafka.binder.consumer-properties.";
+    protected static final String CLIENT_ID = "azure.credential.client-id";
+    protected static final String MANAGED_IDENTITY_ENABLED = "azure.credential.managed-identity-enabled";
+
+    protected final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+            .withConfiguration(AutoConfigurations.of(AzureEventHubsKafkaOAuth2AutoConfiguration.class,
+                    AzureGlobalPropertiesAutoConfiguration.class, AzureTokenCredentialAutoConfiguration.class,
+                    KafkaAutoConfiguration.class));
+
+
+    @Test
+    void shouldNotConfigureWithoutKafkaTemplate() {
+        this.contextRunner
+                .withClassLoader(new FilteredClassLoader(KafkaTemplate.class))
+                .run(context -> assertThat(context).doesNotHaveBean(AzureEventHubsKafkaOAuth2AutoConfiguration.class));
+    }
+
+    @Test
+    void shouldConfigureFactoryCustomizersAndBPP() {
+        this.contextRunner
+                .run(context -> {
+                    assertThat(context).hasSingleBean(DefaultKafkaConsumerFactoryCustomizer.class);
+                    assertThat(context).hasSingleBean(DefaultKafkaProducerFactoryCustomizer.class);
+                    assertThat(context).hasSingleBean(KafkaBinderConfigurationPropertiesBeanPostProcessor.class);
+                });
+    }
+
+    @Test
+    protected abstract void testBindSpringBootKafkaProperties();
+
+    @Test
+    protected abstract void testBindAzureGlobalProperties();
```

Comment:
If you want to create test cases in the abstract class, you can write the test skeleton here and expose an abstract method, and let the subclasses override it.

Reply:
updated

Reply:
please double check